lib/src/main/java/uk/gov/di/ipv/core/library/validation/TokenRequestValidator.java

```diff
@@ -1,37 +1,118 @@
+        this.configurationService = configurationService;
+        this.verifier = getClientAuthVerifier(configurationService);
+    }
+
+    public void authenticateClient(String requestBody, Map<String, String> queryParams)
+            throws ClientAuthenticationException {
+        if (!queryParams.containsKey(CLIENT_ASSERTION_PARAM)) {
+            if (!queryParams.containsKey(CLIENT_ID_PARAM)) {
+                LOGGER.error(
+                        "Missing either client_assertion or client_id values in request. Failed to establish client_id value.");
+                throw new ClientAuthenticationException(
+                        "Unknown client, no client_id value or client_assertion jwt found in request");
+            } else {
+                clientId = queryParams.get(CLIENT_ID_PARAM);
+
+                String clientAuthenticationMethod = getClientAuthenticationMethod(clientId);
 
-            if (claimsSet != null) {
-                return ValidationResult.createValidResult();
+                if (clientAuthenticationMethod.equals(JWT)) {
+                    LOGGER.error("Missing client_assertion jwt for configured client {}", clientId);
+                    throw new ClientAuthenticationException(
+                            String.format(
+                                    "Missing client_assertion jwt for configured client '%s'",
+                                    clientId));
+                }
             }
-        } catch (ParseException e) {
-            LOGGER.error("Unable to parse Claims set {} ", e.getMessage());
-            return new ValidationResult<>(false, OAuth2Error.INVALID_CLIENT);
+        } else {
+            PrivateKeyJWT clientJwt;
+            try {
+                clientJwt = PrivateKeyJWT.parse(requestBody);
+
+                clientId = clientJwt.getClientID().getValue();
+
+                String clientAuthenticationMethod = getClientAuthenticationMethod(clientId);
+
+                if (clientAuthenticationMethod.equals(NONE)) {
+                    return;
+                }
+
+                verifier.verify(clientJwt, null, null);
+                validateExpiration(clientJwt.getJWTAuthenticationClaimsSet());
+            } catch (ParseException | InvalidClientException | JOSEException e) {
+                LOGGER.error("Validation of client_assertion jwt failed");
+                throw new ClientAuthenticationException(e);
+            }
+        }
+    }
+
+    private String getClientAuthenticationMethod(String clientId)
+            throws ClientAuthenticationException {
+        String clientAuthenticationMethod =
+                configurationService.getClientAuthenticationMethod(clientId);
+        if (clientAuthenticationMethod == null) {
+            LOGGER.error("Failed to find config for client ID: {}", clientId);
+            throw new ClientAuthenticationException(
+                    String.format("Config for client ID '%s' not found", clientId));
         }
```

Comment:
The ssmProvider in the config service will throw a runtime error if it can't find a value in the parameter store, so i think this block can be safely dropped.

Reply:
refactor to catch ssmProvider exception so that tests do not fall over